src/helper/parameter.cpp

```diff
@@ -4,172 +4,102 @@
 // This product includes software developed at Datadog
 // (https://www.datadoghq.com/). Copyright 2021 Datadog, Inc.
 #include "parameter.hpp"
-#include "ddwaf.h"
 #include "exception.hpp"
-#include <string>
-#include <string_view>
 
 namespace dds {
 
-parameter::parameter() : _ddwaf_object() { ddwaf_object_invalid(this); }
+parameter::parameter(const ddwaf_object &arg) { *((ddwaf_object *)this) = arg; }
```

Comment:
I think this constructor is a bit dangerous, because it will apply to subclasses of `ddwaf_object`. I see that you deleted copy constructor, but that still leaves parameter_view. In that case, the new `parameter` would take ownership of the object "pointed to" by the parameter_view and invalidate it after destructor, leaving the view dangling.

Maybe having a deleted constructor template for the whole hierarchy would be a good idea:
```cpp
    template<typename T, typename = std::enable_if_t<
        std::conjunction_v<
            std::is_base_of<ddwaf_object, std::remove_cv_t<std::decay_t<T>>>,
            std::negation<std::is_same<ddwaf_object, std::remove_cv_t<std::decay_t<T>>>
        >
    >>>
    parameter(T&& t) = delete;
```